File: libgomp/testsuite/libgomp.fortran/pr90779.f90
! { dg-do run }
! PR middle-end/90779

program pr90779
  implicit none
  integer :: v(4), i

  !$omp target map(from:v)
    v(:) = (/ (i, i=1,4) /)
  !$omp end target

  if (any (v .ne. (/ (i, i=1,4) /))) stop 1
end program

File: libgomp/testsuite/libgomp.c/pr90779.c
/* PR middle-end/90779 */

extern void abort (void);

int
main ()
{
  int i, j;
  for (i = 0; i < 2; ++i)
    #pragma omp target map(from: j)
    {
      static int k = 5;
      j = ++k;
    }
  if (j != 7)
    abort ();
  return 0;
}
